dcicutils/structured_data.py

```diff
@@ -154,7 +154,32 @@ def resolved_refs(self) -> List[str]:
 
     @property
     def resolved_refs_with_uuids(self) -> List[str]:
-        return list([{"path": resolved_ref[0], "uuid": resolved_ref[1]} for resolved_ref in self._resolved_refs])
+        return list([{"path": resolved_ref[0],
+                      "uuid": resolved_ref[1] if len(resolved_ref) >= 2 else None}
+                     for resolved_ref in self._resolved_refs])
+
+    @property
+    def unchecked_refs(self) -> List[str]:
+        """
+        Returns list of unchecked (for existence) references, grouped by reference path;
+        each object in the list has a path property and a srcs property which is a list of
+        src objects containing the type, column and row of the reference to the reference.
+        Note that this is only populated if the norefs option is specified.
+        """
+        def load_json(value: str) -> Optional[dict]:
+            try:
+                return json.loads(value)
+            except Exception:
+                return None
+        result = []
+        if self._norefs:
+            for ref in self._resolved_refs:
+                if len(ref) >= 3 and (ref_path := ref[0]) and (ref_src := load_json(ref[2])):
```

Comment:
Do you mind explaining this? I'm having trouble following this code

Reply:
Yeah this is should be refactored a bit. I've added this comment, hopefully clears it up a bit .. 
# The structure of this self._resolved_refs is setup in Schema._map_function_ref,
# which is called whenever a reference (linkTo) is encountered. It is a set of
# tuples containing three items: [0] the ref path, [1] its uuid (if applicable),
# and [2] its src. The src identifies the place where this ref occurred and is a
# dictionary containing file, type, column, and row properties. For this case, of
# unchecked (norefs) references, the uuid ([1]) is None because we are skipping
# ref resolution. And the src is actually a *string* dump of the dictionary, only
# because dictionarties cannot be put in a set (which is what _resolved_refs is);
# this dump is also done in Schema._map_function_ref (should probably change this
# to be a list to avoid this - TODO); we only even store this src info for this
# norefs case, as not really needed otherwise. This is just to support the
# smaht-submitr/submit-metadata-bundle --info --refs options.